tests/unit/test_utils_filesystem.py

```diff
@@ -59,3 +70,53 @@ def test_copy2_fixed_raises_appropriate_errors(create, error_type, tmpdir):
         copy2_fixed(src, dest)
 
     assert not dest.exists()
+
+
+def test_file_opener_no_file(process):
+    # FileOpener joins the subprocess even if the parent never sends the path
+    with FileOpener():
+        pass
+    assert len(process.children()) == 0
+
+
+def test_file_opener_not_found(tmpdir, process):
+    # The FileOpener cleans up the subprocess when the file cannot be opened
+    path = tmpdir.joinpath('foo.txt')
+    with FileOpener(path):
+        pass
+    assert len(process.children()) == 0
+
+
+def test_file_opener_normal(tmpdir, process):
+    # The FileOpener cleans up the subprocess when the file exists
+    path = tmpdir.joinpath('foo.txt')
+    with open(path, 'w') as f:
+        f.write('Hello\n')
+    with FileOpener(path):
+        pass
+    assert len(process.children()) == 0
+
+
+@skip_unless_windows
+def test_file_opener_produces_unlink_error(tmpdir, process):
```

Comment:
Since these tests are more against Python/Windows than anything in `pip/_internal/utils/filesystem.py`, I would put them in `tests/functional/test_windows.py`.

Reply:
I think moving them to `tests/lib/test_lib_filesystem.py` is hopefully enough. Since producing these errors was the whole point, I'd rather not separate these tests from the other tests.   Please resolve if you think this is OK.